central/apitoken/datastore/datastore_impl.go

```diff
@@ -4,24 +4,54 @@ import (
 	"context"
 
 	"github.com/stackrox/rox/central/apitoken/datastore/internal/store"
+	postgresStore "github.com/stackrox/rox/central/apitoken/datastore/internal/store/postgres"
+	rocksdbStore "github.com/stackrox/rox/central/apitoken/datastore/internal/store/rocksdb"
 	"github.com/stackrox/rox/central/role/resources"
 	v1 "github.com/stackrox/rox/generated/api/v1"
 	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/postgres"
 	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	"github.com/stackrox/rox/pkg/rocksdb"
 	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/search/blevesearch"
 	"github.com/stackrox/rox/pkg/sync"
 )
 
 var (
 	integrationSAC = sac.ForResource(resources.Integration)
+
+	log = logging.LoggerForModule()
 )
 
 type datastoreImpl struct {
-	storage store.Store
+	storage  store.Store
+	searcher search.Searcher
 
 	sync.Mutex
 }
 
+func newPostgres(pool *postgres.DB) *datastoreImpl {
+	storage := postgresStore.New(pool)
+	indexer := postgresStore.NewIndexer(pool)
+	searcher := blevesearch.WrapUnsafeSearcherAsSearcher(indexer)
```

Comment:
curious why we are. using bleve in postgres mode.

Reply:
This `blevesearch` wrapper is a utility from the search package to convert bleve searchers (typically the generated index code) to standard searcher. The difference between the two is the `...blevesearch.SearchOption` parameter in the function signature in the former.

Reply:
fair enough.  If we are going to continue to use that we should probably refactor it at some point.  outside of scope here but worth mentioning.